sdk/ai/Azure.AI.Agents.Persistent/src/Telemetry/OpenTelemetryScope.cs

```diff
@@ -0,0 +1,1159 @@
+        }
+    }
+
+    internal class OpenTelemetryScope : IDisposable
+    {
+        internal enum OpenTelemetryScopeType
+        {
+            Unknown,
+            CreateAgent,
+            CreateThread,
+            CreateMessage,
+            CreateRun,
+            ListMessages,
+            SubmitToolOutputs,
+            ListRunSteps
+        }
+
+        /// <summary>
+        /// Activity source is used to save events and log the tags.
+        /// On the ApplicationInsights events are logged to traces table.
+        /// The tags are not logged, but are shown in the console.
+        /// </summary>
+        private static readonly ActivitySource s_agentSource = new(ClientName);
+        /// <summary>
+        /// Add histograms to log metrics.
+        /// On the ApplicationInsights metrics are logged to customMetrics table.
+        /// </summary>
+        private static readonly Meter s_meter = new Meter(ClientName);
+        private static readonly Histogram<double> s_duration = s_meter.CreateHistogram<double>(
+            GenAiClientOperationDurationMetricName, "s", "Measures GenAI operation duration.");
+        private static readonly Histogram<long> s_tokens = s_meter.CreateHistogram<long>(
+            GenAiClientTokenUsageMetricName, "{token}", "Measures the number of input and output token used.");
+        private static readonly JsonSerializerOptions s_jsonOptions = new JsonSerializerOptions
+        {
+            DefaultIgnoreCondition = JsonIgnoreCondition.WhenWritingNull,
+            Encoder = JavaScriptEncoder.UnsafeRelaxedJsonEscaping // Allow non-ASCII characters
+        };
+
+        private readonly Activity _activity;
+        private readonly Stopwatch _duration;
+        private readonly TagList _commonTags;
+        private static bool s_traceContent = AppContextSwitchHelper.GetConfigValue(
+            TraceContentsSwitch,
+            TraceContentsEnvironmentVariable);
+        private static bool s_enableTelemetry = AppContextSwitchHelper.GetConfigValue(
+            EnableOpenTelemetrySwitch,
+            EnableOpenTelemetryEnvironmentVariable);
+        private RecordedResponse _response;
+        private string _errorType;
+        private Exception _exception;
+
+        private int _hasEnded = 0;
+        private readonly OpenTelemetryScopeType _scopeType;
+        // In your handler class:
+        private StreamingMessage _currentStreamingMessage;
+
+        /// <summary>
+        /// Create the instance of OpenTelemetryScope for agent creation.
+        /// This constructor logs request and starts the execution timer.
+        /// </summary>
+        /// <param name="content">The request options used in the call.</param>
+        /// <param name="endpoint">The endpoint being called.</param>
+        public static OpenTelemetryScope StartCreateAgent(RequestContent content, Uri endpoint)
+        {
+            if (!s_enableTelemetry)
```

Comment:
please add 
```suggestion
            if (!s_enableTelemetry && s_agentSource.HasListeners())
```
and in other places before creating the scope.

We don't want to add any overheads if nobody is actually listening